File: ayrfu-frontend/src/api/services/message.service.js
import axios from '../axios';

const messageService = {
  getAllMessages: async () => {
    return axios.get('/messages');
  },
  
  getMessagesByType: async (type) => {
    return axios.get(`/messages/type/${type}`);
  },
  
  getUnreadMessages: async () => {
    return axios.get('/messages/unread');
  },
  
  getUnreadMessagesByType: async (type) => {
    return axios.get(`/messages/unread/type/${type}`);
  },
  
  getUnreadCount: async () => {
    return axios.get('/messages/unread/count');
  },
  
  getMessageById: async (id) => {
    return axios.get(`/messages/${id}`);
  },
  
  createMessage: async (messageData) => {
    return axios.post('/messages', messageData);
  },
  
  markMessageAsRead: async (id) => {
    return axios.patch(`/messages/${id}/read`);
  },
  
  markMultipleAsRead: async (messageIds) => {
    return axios.post('/messages/mark-read', { messageIds });
  },
  
  deleteMessage: async (id) => {
    return axios.delete(`/messages/${id}`);
  },
  
  searchMessages: async (searchText) => {
    return axios.get(`/messages/search?searchText=${encodeURIComponent(searchText)}`);
  }
};

export default messageService;

File: ayrfu-backend/src/main/java/com/uddan/ayrfu/service/MessageService.java
package com.uddan.ayrfu.service;

import com.uddan.ayrfu.dto.request.MessageRequest;
import com.uddan.ayrfu.dto.response.MessageResponse;
import com.uddan.ayrfu.enumeration.MessageType;
import org.springframework.data.domain.Page;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

public interface MessageService {

    @Transactional
    MessageResponse createMessage(MessageRequest messageRequest);

    @Transactional(readOnly = true)
    MessageResponse getMessageById(Long id);

    @Transactional(readOnly = true)
    List<MessageResponse> getAllMessages(int page, int size);

    @Transactional(readOnly = true)
    List<MessageResponse> getMessagesByType(MessageType type, int page, int size);

    @Transactional(readOnly = true)
    List<MessageResponse> getUnreadMessages();

    @Transactional(readOnly = true)
    Page<MessageResponse> getUnreadMessages(int page, int size);

    @Transactional(readOnly = true)
    List<MessageResponse> getUnreadMessagesByType(MessageType type);

    @Transactional(readOnly = true)
    Page<MessageResponse> getUnreadMessagesByType(MessageType type, int page, int size);

    @Transactional
    MessageResponse markMessageAsRead(Long id);

    @Transactional
    int markMultipleMessagesAsRead(List<Long> messageIds);

    @Transactional
    void deleteMessage(Long id);

    @Transactional(readOnly = true)
    List<MessageResponse> searchMessages(String searchText);

    @Transactional(readOnly = true)
    int getUnreadMessageCount();
}